src/main/java/ch/heigvd/pro/b04/questions/QuestionController.java

```diff
@@ -1,29 +1,192 @@
+  }
+
+  /**
+   * Update a {@link Question} in a {@link ServerPoll}.
+   *
+   * @param token    sender's token
+   * @param question new version of the question
+   * @param idModo   moderator who should be the sender of the request
+   * @param idPoll   poll to add question in
+   * @param maggieQ  id of question to update
+   * @throws ResourceNotFoundException if one parameter is broken
+   */
+  @PutMapping(value = "/mod/{idModerator}/poll/{idPoll}/question/{idQuestion}")
+  public void updateQuestion(@RequestParam(name = "token") String token,
+      @RequestBody Question question,
+      @PathVariable(name = "idModerator") int idModo,
+      @PathVariable(name = "idPoll") ServerPollIdentifier idPoll,
+      @PathVariable(name = "idQuestion") QuestionIdentifier maggieQ)
+      throws ResourceNotFoundException, WrongCredentialsException {
+    testModeratorRight(idModo, idPoll, token);
+    ServerPoll pollTest = pollRepository.findById(idPoll).get();
+    Optional<Question> upQ = repository.findById(maggieQ);
+    if (upQ.isEmpty() || !(pollTest.equals(maggieQ.getIdxPoll()))) {
+      throw new ResourceNotFoundException();
+    }
+
+    upQ.get().setTitle(question.getTitle());
+    upQ.get().setDetails(question.getDetails());
+    upQ.get().setAnswersMax(question.getAnswersMax());
+    upQ.get().setAnswersMin(question.getAnswersMin());
+    upQ.get().setVisibility(question.getVisibility());
+    upQ.get().setIndexInPoll(question.getIndexInPoll());
+  }
+
+  /**
+   * Delete a {@link Question} in a {@link ServerPoll}.
+   *
+   * @param token   sender's token
+   * @param idModo  moderator who should be the sender of the request
+   * @param idPoll  poll to delete question in
+   * @param maggieQ id of question to delete
+   * @throws ResourceNotFoundException if one parameter is broken
+   */
+  @DeleteMapping(value = "DELETE /mod/{idModerator}/poll/{idPoll}/question/{idQuestion}")
+  public void deleteQuestion(@RequestParam(name = "token") String token,
+      @PathVariable(name = "idModerator") int idModo,
+      @PathVariable(name = "idPoll") ServerPollIdentifier idPoll,
+      @PathVariable(name = "idQuestion") QuestionIdentifier maggieQ)
+      throws ResourceNotFoundException, WrongCredentialsException {
+    testModeratorRight(idModo, idPoll, token);
+
+    ServerPoll pollTest = pollRepository.findById(idPoll).get();
+    Optional<Question> upQ = repository.findById(maggieQ);
+    if (upQ.isEmpty() || !(pollTest.equals(maggieQ.getIdxPoll()))) {
+      throw new ResourceNotFoundException();
+    }
+
+    repository.delete(upQ.get());
+  }
+
+  /**
+   * Test if a token belongs to a {@link Moderator}, and if a {@link ServerPoll} belongs to this
+   * Moderator.
+   *
+   * @param idModerator id of the modo to test
+   * @param idPoll      id of the poll
+   * @param token       token of the moderator
+   * @return true it test is correct
+   * @throws ResourceNotFoundException throws exceptions otherwise
```

Comment:
We should have the `@throws WrongCredentialsException` documentation here too.

Reply:
It's corrected